Marlin/src/gcode/feature/trinamic/M911-M915.cpp

```diff
@@ -271,7 +304,7 @@
 /**
  * M914: Set StallGuard sensitivity.
  */
-#if USE_SENSORLESS
+#if USE_SENSORLESS && (X_SENSORLESS || Y_SENSORLESS || Z_SENSORLESS)
```

Comment:
Is there a case where a user would enable sensorless homing/probing but then uncomment all stall sensitivities?

Reply:
this Problem is, where E0, E1, E2 ...  define as TMC2130 and Sensorless give it a unses var warning

Reply:
Then that's a configuration error and should rather be handled in sanitycheck.

Reply:
`const uint8_t index = parser.byteval ('I');`
It would still be unused.

Reply:
Or do you mean the check'if USE_SENSORLESS && (X_SENSORLESS || Y_SENSORLESS || Z_SENSORLESS) `in` SanityCheck.h`

Reply:
`const uint8_t index` requires `USE_SENSORLESS` to be true.
Sanitycheck can force this to be the case only when an applicable driver is enabled on XYZ.

Reply:
> const uint8_t index requires USE_SENSORLESS to be true.

Yes, but if the E0 driver is TMC2130 and XYZ is not, then `index` is unused

Reply:
If E0 is TMC2130 and XYZ is not **and** sensorless homing/probing is **enabled**, you get a sanity check error.

If E0 is TMC2130 and XYZ is not **and** sensorless homing/probing is **disabled**, the code never gets compiled in.

Reply:
#if USE_SENSORLESS  && !(X_SENSORLESS || Y_SENSORLESS || Z_SENSORLESS)
      #error "SENSORLESS_HOMING or SENSORLESS_PROBING is enabled, requires TMC2130 for X or Y or Z"
    #endif